geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -1015,7 +1015,7 @@ public void txApplyDestroy(Object key, TransactionId txId, TXRmtEvent txEvent,
               // a receipt of a TXCommitMessage AND there are callbacks installed
               // for this region
               @Released
-              final EntryEventImpl callbackEvent = EntryEventFactory
+              final EntryEventImpl callbackEvent = new EntryEventFactoryImpl()
```

Comment:
Actually, one more change to make. Please extract the `new EntryEventFactoryImpl()` calls into a private final member field like:
```
private final EntryEventFactory entryEventFactory = new EntryEventFactoryImpl();
```
Otherwise, we'll end up generating a lot of instances for garbage collection to deal with. AbstractRegionMap experiences a lot of traffic such as thousands of operations per second.